services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java

```diff
@@ -46,35 +53,45 @@ public AggRouterHttpRequestStats(
   }
 
   public ScatterGatherStats getScatterGatherStatsForStore(String storeName) {
-    return scatterGatherStatsMap.computeIfAbsent(storeName, k -> new ScatterGatherStats());
+    if (isStoreStatsEnabled) {
+      return scatterGatherStatsMap.computeIfAbsent(storeName, k -> new ScatterGatherStats());
+    } else {
+      return totalScatterGatherStats;
```

Comment:
Will this result in dual recording of the total stats?
if we are checking `isStoreStatsEnabled` and not recording the store stats as seen in `recordStoreStats()`, should we clean this to not return anything?

Reply:
Good catch, removed and rebased.